tools/graph-stats/graph-memory-stats.cpp

```diff
@@ -0,0 +1,308 @@
+/*
+ * License (a copy is located in LICENSE.txt at the top-level directory).
+ *
+ * Copyright (C) 2018, The University of Texas at Austin. All rights reserved.
+ * UNIVERSITY EXPRESSLY DISCLAIMS ANY AND ALL WARRANTIES CONCERNING THIS
+ * SOFTWARE AND DOCUMENTATION, INCLUDING ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR ANY PARTICULAR PURPOSE, NON-INFRINGEMENT AND WARRANTIES OF
+ * PERFORMANCE, AND ANY WARRANTY THAT MIGHT OTHERWISE ARISE FROM COURSE OF
+ * DEALING OR USAGE OF TRADE.  NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH
+ * RESPECT TO THE USE OF THE SOFTWARE OR DOCUMENTATION. Under no circumstances
+ * shall University be liable for incidental, special, indirect, direct or
+ * consequential damages or loss of profits, interruption of business, or
+ * related expenses which may arise from use of Software or Documentation,
+ * including but not limited to those resulting from defects in Software and/or
+ * Documentation, or loss or inaccuracy of data of any kind.
+ */
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+
+#include <arrow/type_traits.h>
+
+#include "katana/ArrowVisitor.h"
+#include "katana/Galois.h"
+#include "katana/JSON.h"
+#include "katana/Logging.h"
+#include "katana/OfflineGraph.h"
+#include "llvm/Support/CommandLine.h"
+#include "tsuba/Errors.h"
+#include "tsuba/FaultTest.h"
+#include "tsuba/FileView.h"
+
+namespace cll = llvm::cl;
+static cll::opt<std::string> inputfilename(
+    cll::Positional, cll::desc("graph-file"), cll::Required);
+
+static cll::opt<std::string> outputfilename(
+    cll::Positional, cll::desc("out-file"), cll::Required);
+
+using map_element = std::unordered_map<std::string, int64_t>;
```

Comment:
@GioLomia : please fix these type names. First: they don't convey what's being stored, e.g., what is a `map_element`? 
Two: You should follow `CamelCase` for all types you define. 
All 3 types below are badly named on both counts.